src/indexmap.rs

```diff
@@ -373,22 +360,19 @@ where
 ///     println!("{}: \"{}\"", book, review);
 /// }
 /// ```
-pub struct IndexMap<K, V, N, S>
+pub struct IndexMap<K, V, S, const N: usize>
 where
     K: Eq + Hash,
-    N: ArrayLength<Bucket<K, V>> + ArrayLength<Option<Pos>>,
 {
     core: CoreMap<K, V, N>,
     build_hasher: S,
 }
 
-impl<K, V, N, S> IndexMap<K, V, N, BuildHasherDefault<S>>
+impl<K, V, S, const N: usize> IndexMap<K, V, BuildHasherDefault<S>, N>
 where
     K: Eq + Hash,
     S: Default + Hasher,
-    N: ArrayLength<Bucket<K, V>> + ArrayLength<Option<Pos>> + PowerOfTwo,
 {
-    // TODO turn into a `const fn`; needs `mem::zeroed` to be a `const fn`
     /// Creates an empty `IndexMap`.
     ///
     /// **NOTE** This constructor will become a `const fn` in the future
```

Comment:
Could this be `const` now? I notice there is a `const-fn` feature of `hash32` to allow the construction of the hasher in `const` time.

Reply:
Hi, this requires an update to `hash32`, the feature is not enough.
I'll look into it.

Reply:
I have made a PR to solve this issue.

https://github.com/japaric/hash32/pull/9